### PR TITLE
Add sensitive option to resources

### PR DESCRIPTION
**Pull request by bmhughes**
### Description
Senstive resource property added to resources to pass through to underlying template and file resources in rule and rule6 to prevent the contents of rule files being logged or output to the console.

### Issues Resolved
- None

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: resources/rule.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@
 property :lines, String
 property :table, Symbol
 property :ipv6, [TrueClass, FalseClass], default: false
+property :sensitive, [TrueClass, FalseClass], default: false
```
Comment (tas50) [inline resources/rule.rb]: This comes on every resource already so you don’t need to define it

File: resources/rule6.rb [hunk 2]
```diff
@@ -22,31 +22,30 @@
 property :variables, Hash, default: {}
 property :lines, String
 property :table, Symbol
+property :sensitive, [TrueClass, FalseClass], default: false
```
Comment (tas50) [inline resources/rule6.rb]: This one can go as well

**Comment by bmhughes** [comment]
Done, removed.

Interestingly is that documented anywhere? As I never realised this before and it can probably remove it from several internal wrapper cookbooks as well. Is it just the sensitive property that is inherited?

I've double checked the custom resource doc page but unless i'm missing it it doesn't seem to be mentioned.

**Comment by bmhughes** [comment]
I've also realised that the generate rule files are world readable so I've added and option to change the mode of the generated files as well. It defaults to the original functionality.

**Comment by tas50** [comment]
Looks good now. Thanks

**Pull request merged**